### ISSUE TITLE
Deadlock risk in -[SPTableData informationForView:] 

### ISSUE THREAD

**Issue by dmoagx** [sequelpro/sequelpro#2175]
Above method can be called from a background thread and tries to display an alert if loading the data fails:
![bildschirmfoto 2015-07-14 um 21 13 58 688](https://cloud.githubusercontent.com/assets/5793864/8682442/478b7ef8-2a6d-11e5-87c4-0e723bd8b055.png)
However the main thread can actually be blocked by the lock the background thread is holding:
![bildschirmfoto 2015-07-14 um 21 15 36 314](https://cloud.githubusercontent.com/assets/5793864/8682486/8d5a797a-2a6d-11e5-9d82-b835a7750982.png)
Therefore creating a deadlock.

This can happen when rapidly switching tables/views while missing permissions for one of them.

**Comment by dmoagx** [comment]
Above fix is not complete:
There is another issue in `[SPTableData tableEncoding]` and others where they can't distinguish "Not loaded" and "Tried loading, but errored" and therefore run in the same error multiple times.
